include/Machine.h

```diff
@@ -0,0 +1,50 @@
+#ifndef _MACHINE_H_
+#define _MACHINE_H_
+
+#include <cstdint>
+#include <array>
+
+#include "FabMember.h"
+
+class Machine
+{
+public:
+  typedef uint32_t MachineIDType;
+  typedef enum enum_MachineType
+  {
+    INVALID = 0x0,
+    PRINTER3D = 0x1,
+    LASER = 0x2,
+    CNC = 0x3,
+    EXTRA1 = 0xFE,
+    EXTRA2 = 0xFF
+  } MachineType;
+
+private:
+  MachineIDType _machine_id = INVALID;
+  MachineType _machine_type = INVALID;
+  uint8_t _control_pin = -1;
+  bool _control_pin_active_low = false;
+
+  bool _active;
+  FabMember _current_user;
+  void _turnOn();
+  void _turnOff();
+  uint32_t _usage_start_timestamp;
+
+public:
+  Machine(MachineIDType machine_id, MachineType machine_type, uint8_t control_pin, bool control_pin_active_low);
+  ~Machine() = default;
+  bool isFree();
+  void begin();
+  FabMember getActiveUser();
+  bool isCardWhitelisted(FabMember user);
```

Comment:
authorization will be handled by server if online, otherwise an hardcoded whitelist is used.

Reply:
ok, so i guess you want an Authorizaton class that given an UID, it will return a Member that tell you if it is valid, and authorized. If he is fishing the auth from a server or from the local array is hidden to the user

Reply:
done as AuthProvider class which manages the whitelist/server queries with a cache.